client/components/PubEdge/PubEdgeEditor.js

```diff
@@ -23,19 +24,56 @@ const PubEdgeEditor = (props) => {
 		dateFormat(publicationDate, 'yyyy-mm-dd'),
 	);
 
+	useEffect(() => {
+		if (publicationDate) {
+			setPublicationDateString(dateFormat(publicationDate, 'yyyy-mm-dd'));
+		}
+	}, [publicationDate]);
+
 	const handlePublicationDateChange = (evt) => {
 		const nextDateString = evt.target.value;
-		setPublicationDateString(nextDateString);
-		onUpdateExternalPublication({ publicationDate: new Date(nextDateString).toString() });
+		onUpdateExternalPublication({
+			publicationDate: nextDateString ? new Date(nextDateString).toString() : null,
+		});
+	};
+
+	const renderPublicationDate = () => {
```

Comment:
Sigh...I can't believe `<input type="date">` is so hard to work with.

Reply:
It's so bad. If you put in an invalid date, its value becomes `''` instead of the incorrectly formatted date, which makes it impossible to provide useful feedback to the user. Same with input type="number".